libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.java

```diff
@@ -93,19 +95,30 @@
     private String userHint;
     private String spActivityName;
     private Bundle spActivityExtras;
+    private AtomicBoolean authConfigFetched;
 
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
 
-        // Getting login options from intent's extras
+        // Getting login options from intent's extras.
         final LoginOptions loginOptions = LoginOptions.fromBundle(getIntent().getExtras());
 
-        // Protect against screenshots
+        // Protect against screenshots.
         getWindow().setFlags(WindowManager.LayoutParams.FLAG_SECURE,
                 WindowManager.LayoutParams.FLAG_SECURE);
 
-        // Setup content view
+        // Fetches auth config if required.
+        authConfigFetched = new AtomicBoolean(false);
+        if (!authConfigFetched.get()) {
+            (new AuthConfigTask(this)).execute();
+        }
+
+        // Waits for auth config to be fetched before continuing to load the activity.
+        while (!authConfigFetched.get()) {
```

Comment:
To avoid the `while()` could you use a BlockingQueue (like we did in [TestRunnerPlugin](https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/master/libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/plugin/TestRunnerPlugin.java#L117) and [JSTestCase](https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/master/libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/util/test/JSTestCase.java#L89))

```java
private BlockingQueue<Boolean> authConfigReady = new ArrayBlockingQueue<Boolean>(1);

// Instead of the while
authConfigReady.take()

// In onAuthConfigFetched
authConfigReady.add(Boolean.TRUE);
```